File: src/main/resources/output/IFC4X3_java/Types/IfcCsgSelect.java
// SELECT TYPE IfcCsgSelect
public class IfcCsgSelect {
    public Object value;

    public IfcCsgSelect(Object value) {
        if (!(value instanceof IfcBooleanResult || value instanceof IfcCsgPrimitive3D)) {
            throw new IllegalArgumentException("Value must be one of: IfcBooleanResult, IfcCsgPrimitive3D");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}

File: src/main/resources/output/IFC4X3_csharp/Types/IfcCsgSelect.cs
// SELECT TYPE IfcCsgSelect
public class IfcCsgSelect {
    public object Value { get; }

    public IfcCsgSelect(object value) {
        if (!(value is IfcBooleanResult || value is IfcCsgPrimitive3D))
        {
            throw new ArgumentException("Value must be one of: IfcBooleanResult, IfcCsgPrimitive3D");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}
